gpgme.go

```diff
@@ -389,7 +390,11 @@ func (c *Context) GetKey(fingerprint string, secret bool) (*Key, error) {
 	key := newKey()
 	cfpr := C.CString(fingerprint)
 	defer C.free(unsafe.Pointer(cfpr))
-	return key, handleError(C.gpgme_get_key(c.ctx, cfpr, &key.k, cbool(secret)))
+	err := handleError(C.gpgme_get_key(c.ctx, cfpr, &key.k, cbool(secret)))
+	if key.k == nil {
```

Comment:
> Is there a specific error that causes this? Could the key be there but you are getting a different error?

Documentation says
> If the key is not found in the keyring, ‘gpgme_get_key’ returns the
> error code ‘GPG_ERR_EOF’ and *R_KEY will be set to ‘NULL’.

So I guess this should test for that exact scenario?

Note that the above contradicts @runcom ’s
> this method simply return `nil, nil` when the key is not found

also looking at https://github.com/projectatomic/atomic/issues/791 , _something_ is creating the “End of file” error reason, and the `GPG_ERR_EOF` does seem to be a likely scenario.

---

So, overall, it seems to me (untested!!) that this should be `if e, ok := err.Error(); key.k == nil && ok && e.(Error).Code() == ErrorEOF` or something like that.